WalletWasabi.Backend/Controllers/BlockchainController.cs

```diff
@@ -416,4 +420,49 @@ private async Task<StatusResponse> FetchStatusAsync(CancellationToken cancellati
 
 		return status;
 	}
+
+	[HttpGet("get-transaction-fee")]
+	[ProducesResponseType(200)]
+	[ProducesResponseType(400)]
+	public async Task<IActionResult> GetTransactionFeeAsync([FromQuery, Required] string transactionId, CancellationToken cancellationToken)
+	{
+		uint256 txID = new(transactionId);
+
+		lock (TransactionFeeCacheLock)
+		{
+			if (TransactionFeeCache.TryGetValue(txID, out Money? value))
+			{
+				return Ok(value);
+			}
+		}
+
+		List<Coin> inputs = new();
+		Dictionary<uint256, Transaction> parentTransactions = new();
+
+		var tx = await RpcClient.GetRawTransactionAsync(txID, true, cancellationToken);
+
+		foreach (var input in tx.Inputs)
+		{
+			if (!parentTransactions.ContainsKey(input.PrevOut.Hash))
+			{
+				var parentTx = await RpcClient.GetRawTransactionAsync(input.PrevOut.Hash, true, cancellationToken).ConfigureAwait(false);
```

Comment:
We can do that, but for me it feels like a bit too much caching. 
It would fill up quickly and I'm not sure how useful would it be.

For example: 

- We request a CJ txid
- ~100 parentTx gets cached (remixes share their parent tx)
- When will these ~100 Tx be used again?

I think the local cache is enough, but I can be convinced easily.

Reply:
CJ A has inputs coming from 10 different coinjoin rounds in total. CJ B also from 10 different coinjoin rounds. They will overlap. All CJs will be parent of remixes in many different rounds.

We can keep that for another PR, same as https://github.com/zkSNACKs/WalletWasabi/pull/11679#discussion_r1400107293

Reply:
I will implement this here quickly.

> they will overlap, and without caching we will need to request several time.

Yes, we can save Rpc requests, which is always good.